### PR TITLE
Gague display & Reset button

### PR DESCRIPTION
**Pull request by Seyi-Toluhi**
Gauge display component set up and is set to render as 1 to allow us test reset button.
Reset button changes energy gauge to 0.
Tests written to check that gauge display is rendered as 1 initially, reset button is rendered on dashboard and when clicked, gauge display is reset to 0.

### CODE REVIEW

File: Frontend/package.json [hunk 1]
```diff
@@ -7,11 +7,17 @@
     "dev": "vite",
     "build": "vite build",
     "lint": "eslint . --ext js,jsx --report-unused-disable-directives --max-warnings 0",
-    "preview": "vite preview"
+    "preview": "vite preview",
+    "test": "vitest"
   },
   "dependencies": {
+    "@testing-library/react": "^15.0.4",
```
Comment (tomgame984) [inline Frontend/package.json:14]: In @DeviloperG PR he refers to `"^15.0.3"`

Could this be a potential conflict when merging branches?

File: Frontend/tests/components/Gague.test.jsx
Comment (tomgame984) [inline Frontend/tests/components/Gague.test.jsx:1]: **Additional Unit Test Cases:**
We see these tests as essential to validate the reset button when the gauge is in both +/- states and with a value >1. 

1. When the Dashboard is loaded and the counter value is set at -1, when the reset button is pressed, the counter is reset to 0.

2. When the Dashboard is loaded and the counter value is set at -10, when the reset button is pressed, the counter is reset to 0.

3. When the Dashboard is loaded and the counter value is set at +10, when the reset button is pressed, the counter is reset to 0.

**Review by tomgame984 — DISMISSED** [review]
We have reviewed tests cases and identify an additional 3 cases that we feel are key to ensure reset button functionality.

File: Frontend/package.json (hunk 1 above)
Comment (Seyi-Toluhi) [reply Frontend/package.json:14]: yes i think this might cause a conflict. i'll have a chat with firat  to see if he can update his version.

Comment (Seyi-Toluhi) [reply Frontend/package.json:14]: Actually, we can just choose my version when merging.

File: Frontend/tests/components/Gague.test.jsx
Comment (Seyi-Toluhi) [reply Frontend/tests/components/Gague.test.jsx:1]: We think these additional tests should come in when we have merged the positive and negative buttons to change the value of counter dynamically. Without these buttona, we had to hardcode the value of counter to 1. once we merge both PR and refactor our codes to integrate all aspects on the dashboard, we will get these additional tests written.

**Review by tomgame984 — APPROVED** [review]
PR approved on provision that additional unit tests to validate reset button to be added once merged to main branch and integration setup.

**Review by F-Gulmez — APPROVED** [review]

**Pull request merged**